lib/bricks-net-api-types.js

```diff
@@ -0,0 +1,1176 @@
+			_this.resolver.resolve(_this);
+		}
+		
+		var promise = when.promise(function (resolve, reject) {
+			_this._DEBUG_LOG('HTTPClient#_performRequest: Connect:',
+				_this.parsed_url_.host, _this.parsed_url_.port);
+			
+			socket = new net.Socket();
+			
+			socket.connect({
+				host:  _this.parsed_url_.host,
+				port:  _this.parsed_url_.port
+			}, function (err) {
+				_this._DEBUG_LOG('HTTPClient#_performRequest: Connect callback:', err || 'OK');
+				
+				if (err) { return reject(err); }
+				resolve();
+			});
+		});
+		
+		function writeNext() {
+			var chunk = writes.shift();
+			
+			return when.promise(function (resolve, reject) {
+				_this._DEBUG_LOG('HTTPClient#_performRequest: Write:', JSON.stringify(chunk));
+				
+				socket.write(chunk, 'utf8', function (err) {
+					_this._DEBUG_LOG('HTTPClient#_performRequest: Write callback:', err || 'OK');
+					
+					if (err) { return reject(err); }
+					resolve();
+				});
+			});
+		}
+		
+		writes.forEach(function () {
+			promise = promise.then(writeNext);
+		});
+		
+		promise = promise.then(function () {
+			_this._DEBUG_LOG('HTTPClient#_performRequest: new HTTPRedirectableRequestData');
+			
+			_this.http_request_ = new HTTPRedirectableRequestData(socket);
+			
+			return _this.http_request_;
+		});
+		
+		promise = promise.then(function () {
+			var response_code_as_int = parseInt(_this.http_request_.RawPath(), 10);
+			
+			_this._DEBUG_LOG('HTTPClient#_performRequest: Parsing response code:', response_code_as_int);
+			
+			_this.response_code_ = HTTPResponseCode(response_code_as_int);
+			
+			if (response_code_as_int >= 300 && response_code_as_int <= 399 && _this.http_request_.location) {
+				_this._DEBUG_LOG('HTTPClient#_performRequest: Got a redirect:', _this.http_request_.location);
+				
+				// Note: This is by no means a complete redirect implementation.
+				_this.parsed_url_ = URL(_this.http_request_.location, _this.parsed_url_);
+				_this.response_url_after_redirects_ = _this.parsed_url_.ComposeURL();
+				
+				// Close the current socket before sending the next request.
+				cleanup();
+				
+				return _this._performRequest();
+			}
+			else {
+				finish();
+			}
+		});
+		
+		return promise;
+	};
+}
+HTTPClient.prototype.Go = function () {
+	var _this = this;
+	
+	// Lazy evaluation.
+	if (!_this._requested) {
+		_this._requested = true;
+		
+		_this.response_url_after_redirects_ = _this.request_url_;
+		_this.parsed_url_ = URL(_this.request_url_);
+		_this.all_urls_ = {};
+		
+		_this._DEBUG_LOG('HTTPClient#Go:', _this.parsed_url_.ComposeURL());
+		
+		_this._performRequest();
+	}
+	
+	// The return value `true`/`false` can only be a promise in async implementation.
+	return when(_this).then(function () {
+		return true;
+	}, function (err) {
+		return false;
+	});
+};
+HTTPClient.prototype.HTTPRequest = function () {
+	return this.http_request_;
+};
+
+
+function PrepareInput(request, client) {
+	// The `request` below is actually `request_params`.
```

Comment:
Because you named it `request` in Bricks code :)

This method covers two signatures of the corresponding C++ method which has got different names for the same argument of different type (and that's OK but I cannot do this in JavaScript without using `cpp-arguments` which I wanted to avoid for non-API-exported code). Note that below the same argument is commented as being `response_params`.